docs/website/docs/plus/getting-started/transformations.md

```diff
@@ -0,0 +1,446 @@
+@dlt_plus.transform(transformation_type="sql", lineage_mode="best_effort", columns={"static_column": {"type": "double"}})
+def mutated_transformation(dataset: SupportsReadableDataset):
+    return dataset.employees.mutate(static_column=1234)
+
+```
+
+## SQL vs Python transformations
+
+As you have seen above, you can choose to execute any transformation either as a Python or SQL transformation. For both transformation types, the basis is a decorated function that returns something that can be evaluated into a SQL query, either a string or a ReadableIbisRelation object. Both transformation types can be mixed and they both result in new tables in the destination dataset with a schema derived from the source dataset. The key differences are:
+
+Python transformations
+* Can be run from any dataset to any destination dataset, you only need to ensure that the destination supports all column types you are using in the transformation or clean them up
+* Will run the select query on the source dataset but extract all resulting rows into the pipeline as Arrow tables / Parquet files and push them into the destination
+* Chunk size for Arrow table extract can be set with the `chunk_size` parameter on the dlt.transform decorator: `@dlt.transform(transformation_type="python", chunk_size=1000)`
+* When instantiated, behaves like a regular dlt.resource - you can pipe it into a dlt.transformer, add filters and maps with 'add_filter' and 'add_map' methods
+* Is generally much slower than pure SQL transformations
+* Will also work if not all resulting columns can be traced to origin columns (unless specified otherwise, more on lineage below)
+
+SQL transformations
+* Can only be run if the source and destination dataset are the same or on the same physical destination. Currently there is no proper check, so using SQL in a place where you should not will result in an obscure error
+* Are much faster than Python transformations and should be used if the use case allows for it
+* Only work if the full resulting schema is known ahead - this means lineage must pass in strict mode. Alternatively, you can supply column hints for columns that lineage cannot trace to the origin. This is the case, for example, if you add a static column in a .mutate() call on a ReadableIbisRelation
+* You cannot pipe the transformation into a dlt.transformer or do any kind of other data manipulation on dataitems, since the data never leaves the database
+
+
+## Sync Source
+As you have observed above, there are cases where you essentially would like to copy a table from one dataset to another but do not really need to build any SQL expressions, because you do not want to filter or join. Let's say you simply want to copy the full company dataset into the warehouse dataset without any kind of data cleanup - you can very easily do this with the sync source. Under the hood, the sync source is a regular dlt source with a list of `dlt.transform` objects that run this sync.
```

Comment:
I think if we publish this as a public tutorial we should from the beginning describe the recommended methods for doing certain things. So use the `sync_source` function from the beginning of the tutorial